cura/Settings/MachineManager.py

```diff
@@ -1650,3 +1658,50 @@ def getAbbreviatedMachineName(self, machine_type_name: str) -> str:
                 abbr_machine += stripped_word
 
         return abbr_machine
+
+    def getMachineTypeNameFromId(self, machine_type_id: str) -> str:
+        machine_type_name = ""
+        results = self._container_registry.findDefinitionContainersMetadata(id = machine_type_id)
+        if results:
+            machine_type_name = results[0]["name"]
+        return machine_type_name
+
+    @pyqtSlot(QObject)
+    def associateActiveMachineWithPrinterDevice(self, printer_device: Optional["PrinterOutputDevice"]) -> None:
```

Comment:
I don't feel that this should be here. It adds all kind of knowledge about the output device (and it's Ultimaker specific details) to the machine manager.

Reply:
Moved to UM3OutputDevicePlugin in d34b3b858